plugin/ochttp/propagation/tracecontext/propagation.go

```diff
@@ -98,13 +104,26 @@ func (f *HTTPFormat) SpanContextFromRequest(req *http.Request) (sc trace.SpanCon
 	return sc, true
 }
 
+func getRequestHeader(req *http.Request, name string, commaSeparated bool) (hdr string, ok bool) {
+	ok = false
+	for k, v := range req.Header {
+		if strings.EqualFold(k, name) {
+			if commaSeparated || len(v) == 1 {
+				ok = true
+			}
+			return strings.Join(v, ","), ok
+		}
+	}
+	return "", false
+}
+
 // TODO(rghetia): return an empty Tracestate when parsing tracestate header encounters an error.
 // Revisit to return additional boolean value to indicate parsing error when following issues
 // are resolved.
 // https://github.com/w3c/distributed-tracing/issues/172
 // https://github.com/w3c/distributed-tracing/issues/175
 func tracestateFromRequest(req *http.Request) *tracestate.Tracestate {
-	h := req.Header.Get(tracestateHeader)
+	h, _ := getRequestHeader(req, tracestateHeader, true)
```

Comment:
It's kind of weird that we join headers with commas only to then split by comma again below (line 131). Seems like we could avoid this unnecessary back and forward conversion by keeping the headers separate and splitting each one separately and then concatenating all the results after splitting.

Reply:
I intended to keep the HTTP layer and TraceContext separate.
Combining duplicated headers using COMMA is mentioned by the HTTP RFC.
Splitting the COMMA separated key-value pair is required by W3C TraceContext.